### PR TITLE
Mesos Tap service main entry point

### PR DESCRIPTION
**Pull request by pallavidn**

### CODE REVIEW

File: cmd/conf-files/apache-mesos-target-conf-template.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "master": "Apache Mesos",
+  "master-ip": "<Mesos Master IP>",
+  "master-port": "",
```
Comment (DongyiYang) [inline cmd/conf-files/apache-mesos-target-conf-template.json]: does user need to specify port?

File: cmd/conf-files/container-conf-template.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "serverMeta": {
+    "turboServer": "https://<turbo-server-ipaddress>:<turbo-server-port>"
+  },
+  "restAPIConfig": {
+    "opsManagerUserName": "administrator",
+    "opsManagerPassword": "admin"
```
Comment (DongyiYang) [inline cmd/conf-files/container-conf-template.json]: change to \<username\> and \<password\>

File: cmd/service/mesosturbo_service.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	var turboCommConfigData *service.TurboCommunicationConfig
+	var turboConfErr error
+	if turboCommConf != "" {
+		turboCommConfigData, turboConfErr = service.ParseTurboCommunicationConfig(turboCommConf)
+	} else {
+		wsConfig := mediationcontainer.WebSocketConfig {
+			LocalAddress:    "http://127.0.0.1",
+			WebSocketUsername:   "vmtRemoteMediation",
+			WebSocketPassword: "vmtRemoteMediation",
+			WebSocketPath: "/vmturbo/remoteMediation",
```
Comment (DongyiYang) [inline cmd/service/mesosturbo_service.go]: We'd better not set any default value here. Actually, the default value for web socket config is set during validation, [here](https://github.com/turbonomic/turbo-go-sdk/blob/37cb7b066b91c00759fcccaea2ce72ca02e503b1/pkg/mediationcontainer/mediation_communication_config.go#L40).

File: cmd/service/mesosturbo_service.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+	var mesosConfErr error
+	if targetConf != "" {
+		mesosTargetConf, mesosConfErr = conf.NewMesosTargetConf(targetConf)
+	} else {
+		var master conf.MesosMasterType
+		if s.Master == string(conf.Apache) {
+			master = conf.Apache
+		} else if s.Master == string(conf.DCOS){
+			master = conf.DCOS
+		}
```
Comment (DongyiYang) [inline cmd/service/mesosturbo_service.go]: an else is needed to indicate an invalid or unsupported MesosMasterType

File: cmd/service/mesosturbo_service.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+		os.Exit(1)
+	}
+
+	// ============================================
+	// Mesos Probe Registration Client
+	registrationClient := mesos.NewRegistrationClient(mesosMasterType)
+
+	// Mesos Probe Discovery Client
+	discoveryClient, err := discovery.NewDiscoveryClient(mesosMasterType, mesosTargetConf)
+
```
Comment (DongyiYang) [inline cmd/service/mesosturbo_service.go:139]: remove this empty line.

File: cmd/service/mesosturbo_service.go [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+	if err != nil {
+		glog.Errorf("Error creating discovery client for "+string(mesosMasterType)+"::"+mesosTargetConf.MasterIP+"\n", err.Error())
+		os.Exit(1)
+	}
+
+	target1 := mesosTargetConf.MasterIP
```
Comment (DongyiYang) [inline cmd/service/mesosturbo_service.go]: use another name instead of target1

File: cmd/service/mesosturbo_service.go [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+	tapService, err :=
+		service.NewTAPServiceBuilder().
+			WithTurboCommunicator(turboCommConfigData).
+			WithTurboProbe(probe.NewProbeBuilder(string(mesosMasterType), probeCategory).
+			RegisteredBy(registrationClient).
+			DiscoversTarget(target1, discoveryClient)).
+			Create()
+
```
Comment (DongyiYang) [inline cmd/service/mesosturbo_service.go]: remove empty line.

File: cmd/service/mesosturbo_service.go [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+	if err != nil {
+		glog.Errorf("Error creating TAP Service : ", err)
+	}
+	//
```
Comment (DongyiYang) [inline cmd/service/mesosturbo_service.go]: remove "//"

File: cmd/mesosturbo.go [hunk 9]
```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	goflag "flag"
+	"runtime"
+	"github.com/turbonomic/mesosturbo/cmd/service"
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
```
Comment (songbinliu) [inline cmd/mesosturbo.go:8]: Package pflag is a drop-in replacement for Go's flag package.
Is the goflag still in use?

**Review by songbinliu — APPROVED** [review]

**Pull request merged**